optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/util/DateTimeUtils.java

```diff
@@ -1,23 +1,173 @@
 package org.optaplanner.openshift.employeerostering.gwtui.client.util;
 
 import java.time.Duration;
+import java.time.Instant;
 import java.time.LocalDate;
+import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.zone.ZoneRules;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsType;
+import org.optaplanner.openshift.employeerostering.gwtui.client.tenant.TenantStore;
+import org.optaplanner.openshift.employeerostering.shared.common.GwtJavaTimeWorkaroundUtil;
+
+@Singleton
 public class DateTimeUtils {
```

Comment:
This class contains some logic, partly in native part that will be compiled into JS. Is there any way of testing it?

Reply:
I think a test.js file came with the moment.js file from the mvn repo; I can make it inject something which is just an interface to moment, and we can inject something else to make sure it is using the right api calls